GVRf/Framework/src/org/gearvrf/utility/Preconditions.java

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2015 Samsung Electronics Co., LTD
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gearvrf.utility;
+
+public final class Preconditions {
```

Comment:
> the Preconditions class makes its constructor private 

Yes, explicitly coding a private constructor is the Java Way to do it. Declaring a class `abstract` sounds like it is meant to be inherited from.

But when I think of all the meanings of `final`, overloading `abstract` sure seems very Java like. And, it's *much* less  code.

>  simple and not used that frequently so that I made it final. 

`final` means you can't inherit from it, not that you can never change the code: I have gotten into more trouble by making things `final` than by not. We (or, maybe, Brazil) will probably grow this class slowly, as we need new tests. Who knows, maybe at some point there will be some sort of instance state and/or methods that people _will_ want to be polymorphic. At that point, we'd have to remove the `final` that we  really shouldn't be adding now.